ci: Add neo4j config instructions (#457)
GoogleCloudPlatform/genai-databases-retrieval-app: docs/datastore/neo4j_graph.md

# Setup and configure Neo4j

## Before you begin

1. Make sure you have a GoogleCloud project and billing is enabled.

1. Set your `PROJECT_ID` environment variable:

    ```bash
    export PROJECT_ID=<YOUR_PROJECT_ID>
    ```

1. [Install](https://cloud.google.com/sdk/docs/install) the gcloud CLI.

1. Set gcloud project:

    ```bash
    gcloud config set project $PROJECT_ID
    ```

1. Make sure you have python version 3.11+ installed.

    ```bash
    python -V
    ```

1. [Install python][install-python] and set up a python [virtual environment][venv].

1. Install required dependencies:
    ```bash
    cd retrieval_service
    pip install -r requirements.txt
    ```

## Create a Neo4j database

> 1. In the [Neo4j console](https://console.neo4j.io/?product=aura-db&_gl=1*mpkyd1*_ga*NzM2MTEwNTQ4LjE3MTk1MDkyNTA.*_ga_DZP8Z65KK4*MTcyMjYyNzYyOC41OC4xLjE3MjI2MzQyODMuMC4wLjA.*_gcl_aw*R0NMLjE3MjI1NTA5NTcuQ2p3S0NBanc1S3kxQmhBZ0Vpd0E1akd1anR2c0FLbVdqVXB3SXNHTE5VQkEzcjh4Zm9WSjk5ZkdXdnl1UEM4bHI4YmZGSUVmMkM4NTF4b0NVOHNRQXZEX0J3RQ..*_gcl_au*MTY0NDkyNTM4OS4xNzE5NTA5MjUwLjE4NjQyNTUxNDguMTcyMTM3NDE3NS4xNzIxMzc0MTc0*_ga_DL38Q8KGQC*MTcyMjYyNzYyOC41Mi4xLjE3MjI2MzQyODMuMC4wLjA.), click `New Instance`, then follow the on-screen instructions to download the credentials as a .txt

Reviewer: ```suggestion
1. In the [Neo4j console](https://console.neo4j.io/?product=aura-db&_gl=1*mpkyd1*_ga*NzM2MTEwNTQ4LjE3MTk1MDkyNTA.*_ga_DZP8Z65KK4*MTcyMjYyNzYyOC41OC4xLjE3MjI2MzQyODMuMC4wLjA.*_gcl_aw*R0NMLjE3MjI1NTA5NTcuQ2p3S0NBanc1S3kxQmhBZ0Vpd0E1akd1anR2c0FLbVdqVXB3SXNHTE5VQkEzcjh4Zm9WSjk5ZkdXdnl1UEM4bHI4YmZGSUVmMkM4NTF4b0NVOHNRQXZEX0J3RQ..*_gcl_au*MTY0NDkyNTM4OS4xNzE5NTA5MjUwLjE4NjQyNTUxNDguMTcyMTM3NDE3NS4xNzIxMzc0MTc0*_ga_DL38Q8KGQC*MTcyMjYyNzYyOC41Mi4xLjE3MjI2MzQyODMuMC4wLjA.), click `New Instance`, then follow the on-screen instructions to download the credentials as a .txt file.
```

# Setup and configure Neo4j

## Before you begin

1. Make sure you have a GoogleCloud project and billing is enabled.

1. Set your `PROJECT_ID` environment variable:

    ```bash
    export PROJECT_ID=<YOUR_PROJECT_ID>
    ```

1. [Install](https://cloud.google.com/sdk/docs/install) the gcloud CLI.

1. Set gcloud project:

    ```bash
    gcloud config set project $PROJECT_ID
    ```

1. Make sure you have python version 3.11+ installed.

    ```bash
    python -V
    ```

1. [Install python][install-python] and set up a python [virtual environment][venv].

1. Install required dependencies:
    ```bash
    cd retrieval_service
    pip install -r requirements.txt
    ```

## Create a Neo4j database

> 1. In the [Neo4j console](https://console.neo4j.io/?product=aura-db&_gl=1*mpkyd1*_ga*NzM2MTEwNTQ4LjE3MTk1MDkyNTA.*_ga_DZP8Z65KK4*MTcyMjYyNzYyOC41OC4xLjE3MjI2MzQyODMuMC4wLjA.*_gcl_aw*R0NMLjE3MjI1NTA5NTcuQ2p3S0NBanc1S3kxQmhBZ0Vpd0E1akd1anR2c0FLbVdqVXB3SXNHTE5VQkEzcjh4Zm9WSjk5ZkdXdnl1UEM4bHI4YmZGSUVmMkM4NTF4b0NVOHNRQXZEX0J3RQ..*_gcl_au*MTY0NDkyNTM4OS4xNzE5NTA5MjUwLjE4NjQyNTUxNDguMTcyMTM3NDE3NS4xNzIxMzc0MTc0*_ga_DL38Q8KGQC*MTcyMjYyNzYyOC41Mi4xLjE3MjI2MzQyODMuMC4wLjA.), click `New Instance`, then follow the on-screen instructions to download the credentials as a .txt file

Reviewer: Can you also add instruction to set up neo4j on GCP?

# Setup and configure Neo4j

## Before you begin

> 1. Make sure you have a GoogleCloud project and billing is enabled.

Reviewer 2: ```suggestion
1. Make sure you have a Google Cloud project and billing is enabled.
```

1. Set your `PROJECT_ID` environment variable:

    ```bash
    export PROJECT_ID=<YOUR_PROJECT_ID>
    ```

1. [Install](https://cloud.google.com/sdk/docs/install) the gcloud CLI.

1. Set gcloud project:

    ```bash
    gcloud config set project $PROJECT_ID
    ```

> 1. Make sure you have python version 3.11+ installed.

Reviewer 2: You still need to enable the AIPlatform API, see the other instructions for the command

    ```bash
    python -V
    ```

> 1. [Install python][install-python] and set up a python [virtual environment][venv].

Reviewer 2: Need to add the links like:
```
[install-python]: https://cloud.google.com/python/docs/setup#installing_python
[venv]: https://cloud.google.com/python/docs/setup#installing_and_using_virtualenv
```

1. Install required dependencies:
    ```bash
    cd retrieval_service

>     pip install -r requirements.txt

Reviewer 2: @Yuan325  in the future we may want to separate the requirements for each db into separate files

    ```

## Create a Neo4j database

> 1. Important: Enabling Neo4j Aura in Google Cloud Marketplace does not permit the use of `Free-tier` instances. It will provision a `Professional-tier` instance, which incurs additional monthly costs.

Reviewer 2: ```suggestion
1. ⚠️**Important**: Enabling [Neo4j Aura in Google Cloud Marketplace](https://pantheon.corp.google.com/marketplace/product/endpoints/prod.n4gcp.neo4j.io) does not permit the use of `Free-tier` instances. It will provision a `Professional-tier` instance, which incurs additional monthly costs. Learn more at [Neo4j AuraDB overview](https://neo4j.com/docs/aura/auradb/?utm_source=gcp).
```

> 1. Important: Enabling Neo4j Aura in Google Cloud Marketplace does not permit the use of `Free-tier` instances. It will provision a `Professional-tier` instance, which incurs additional monthly costs.

Reviewer: Let's create two sub-sections and have instructions under the corresponding section. Also for any attached link 

### Neo4j AuraDB on GCP
you can choose not to include link for this option as it is auto-directed and has a lot of extra parameters. The link that you currently attached has a bunch of session data that shouldn't be included.

### Neo4j Aura (free)
show steps and link https://console.neo4j.io/ to this option

> 1. Important: Enabling Neo4j Aura in Google Cloud Marketplace does not permit the use of `Free-tier` instances. It will provision a `Professional-tier` instance, which incurs additional monthly costs.

Reviewer: ```suggestion

```

> 1. Enable Neo4j AuraDB in GCP Marketplace [Neo4j console](https://console.cloud.google.com/marketplace/product/endpoints/prod.n4gcp.neo4j.io?authuser=2). After enabling it, a `Manage on Provider` button will appear. Click on it to proceed.

Reviewer 2: ```suggestion
1. Enable [Neo4j AuraDB in Google Cloud Marketplace](https://console.cloud.google.com/marketplace/product/endpoints/prod.n4gcp.neo4j.io). After enabling it, a `Manage on Provider` button will appear. Click on it to proceed.
```

> 1. In the [Neo4j console](https://console.neo4j.io/?ref=aura-lp&mpp=4bfb2414ab973c741b6f067bf06d5575&mpid=%24device%3A19050fa4558249-004c1750b934b6-6a3f4f73-18f258-19050fa4558249&_gl=1*9tzqtf*_ga*NzM2MTEwNTQ4LjE3MTk1MDkyNTA.*_ga_DZP8Z65KK4*MTcyNDA5OTc0OS44NC4xLjE3MjQxMDMwMTkuMC4wLjA.*_gcl_aw*R0NMLjE3MjQxMDMwMTkuQ2owS0NRancyb3UyQmhDQ0FSSXNBTkF3TTJIWGhNang4U3l3VEJDYVh6NWZHSWd6Y2VmTHN0dTJoTjJnTXpKd2ZEN1B1U2J1RDdLNFA0OGFBbVRjRUFMd193Y0I.*_gcl_au*MTY0NDkyNTM4OS4xNzE5NTA5MjUwLjE4NjQyNTUxNDguMTcyMTM3NDE3NS4xNzIxMzc0MTc0*_ga_DL38Q8KGQC*MTcyNDA5OTc0OS43OC4xLjE3MjQxMDMwMjAuMC4wLjA.), click `New Instance`, then choose `Professional Tier`. Finally follow the on-screen instructions to download the credentials as a .txt file

Reviewer 2: ```suggestion
1. In the [Neo4j console](https://console.neo4j.io/?ref=aura-lp), click `New Instance`, then choose `Professional Tier`. Finally follow the on-screen instructions to download the credentials as a TXT file.
```

Reviewer 2: Can you confirm the short link works? You may also want to add a note that they will need to create a Neo4j account

Author: They don't need to create an account previously with Neo4j in GCP,  the account gets created automatically. However if they do it outside GCP they will need to create one.  I will specify this.

## Update config

> Update `config.yml` with your database information.

Reviewer 2: You need to add the step to make a copy first

```bash
host: 0.0.0.0
datastore:
  kind: "neo4j"
  uri: "neo4j_url"
  auth:
    username: "neo4j_user"
    password: "neo4j_password"
```

## Initialize data in Neo4j

1. Change to the `retrieval_service` directory:

    ```bash

>     cd retrieval_service

Reviewer 2: You've already changed into this directory above. You may wish to just install the dependencies here or remove this step

    ```

2. Populate your Neo4j database with the command below. It will take several minutes to run:

    ```bash
    python run_database_init.py
    ```

## Developer information

This section is for developers that want to develop and run the app locally.

### Test Environment Variables

Set environment variables:

```bash
export DB_URI=""
export DB_USER=""
export DB_PASSWORD=""
```

Run retrieval service unit tests:

```bash
gcloud builds submit --config retrieval_service/neo4j_graph.test.cloudbuild.yaml \

>     --substitutions _DATABASE_URI=$DB_URI

Reviewer 2: What about user and password?

Author: I used as an example AlloyDB and they only used public variables for substitution. I double checked by running the cloud command and everything worked fine.
